scripts/utils/price-utils.js

```diff
@@ -1,13 +1,96 @@
+  const checkCorrectnessOfDeposits = async (
+    currentPrice,
+    bracketAddress,
+    exchange,
+    stableToken,
+    targetToken,
+    investmentStableTokenPerBracket,
+    investmentTargetTokenPerBracket
+  ) => {
+    const bracketExchangeBalanceStableToken = (await exchange.getBalance(bracketAddress, stableToken.address)).toString()
+    const bracketExchangeBalanceTargetToken = (await exchange.getBalance(bracketAddress, targetToken.address)).toString()
+    const auctionElements = exchangeUtils.decodeOrdersBN(await exchange.getEncodedUserOrders.call(bracketAddress))
+    const bracketOrders = auctionElements.filter(order => order.user.toLowerCase() == bracketAddress.toLowerCase())
+    const stableTokenId = await exchange.tokenAddressToIdMap.call(stableToken.address)
+    const sellStableTokenOrder = bracketOrders.filter(order => order.sellToken == stableTokenId)[0]
+
+    const targetTokenId = await exchange.tokenAddressToIdMap.call(targetToken.address)
+    const sellTargetTokenOrder = bracketOrders.filter(order => order.sellToken == targetTokenId)[0]
+
+    // Check that tokens with a lower price than the target price are funded with stableTokens
+    if (checkThatOrderPriceIsBelowTarget(currentPrice, sellStableTokenOrder)) {
+      // checks whether price is in middle of bracket:
+      if (checkThatOrderPriceIsBelowTarget(1 / currentPrice, sellTargetTokenOrder)) {
+        assert.isTrue(
+          checkFundingInTheMiddleBracket(
+            bracketExchangeBalanceStableToken,
+            bracketExchangeBalanceTargetToken,
+            investmentTargetTokenPerBracket,
+            investmentStableTokenPerBracket
+          )
+        )
+      } else {
+        assert.equal(bracketExchangeBalanceStableToken, investmentStableTokenPerBracket.toString())
+      }
+    } else {
+      assert.equal(bracketExchangeBalanceStableToken, 0)
+    }
+
+    if (checkThatOrderPriceIsBelowTarget(1 / currentPrice, sellTargetTokenOrder)) {
+      if (checkThatOrderPriceIsBelowTarget(currentPrice, sellStableTokenOrder)) {
+        assert.isTrue(
+          checkFundingInTheMiddleBracket(
+            bracketExchangeBalanceStableToken,
+            bracketExchangeBalanceTargetToken,
+            investmentTargetTokenPerBracket,
+            investmentStableTokenPerBracket
+          )
+        )
+      } else {
+        assert.equal(bracketExchangeBalanceTargetToken, investmentTargetTokenPerBracket.toString())
+      }
+    } else {
+      assert.equal(bracketExchangeBalanceTargetToken, 0)
+    }
+  }
+
+  const checkThatOrderPriceIsBelowTarget = function(currentPrice, order) {
+    const multiplicator = 1000000000
+    return new BN(currentPrice * multiplicator).mul(order.priceNumerator) > order.priceDenominator.mul(new BN(multiplicator))
+  }
+
+  const checkFundingInTheMiddleBracket = function(
+    bracketExchangeBalanceStableToken,
+    bracketExchangeBalanceTargetToken,
+    investmentTargetTokenPerBracket,
+    investmentStableTokenPerBracket
```

Comment:
The order of the input values is a bit strange, is there a reason why this is not:
```
    bracketExchangeBalanceStableToken,
    bracketExchangeBalanceTargetToken,
    investmentStableTokenPerBracket,
    investmentTargetTokenPerBracket
```